pkg/webservice/state_dump.go

```diff
@@ -157,7 +160,20 @@ func doStateDump(w io.Writer) error {
 	if err != nil {
 		return err
 	}
-	if _, err = w.Write(prettyJSON); err != nil {
+
+	stateLog := log.New(w, "", 0)
```

Comment:
do we need to new a state log every time?
should we just reuse the same object and use it to write the state data over time?

Reply:
Good point -- this may be wasteful. However, the log object requires being constructed from the response writer which is why I had it in this method. I'll familiarize myself more with how the response writer gets passed from `routes.go` and see if I can change this.

Reply:
@yangwwei Maybe I'm missing something obvious but I couldn't get this working after several hours. I tried to use `zapcore` directly in `webservices.go` so we only instantiate one logger in `NewWebApp` as per the [zap/lumberjack documentation](https://github.com/uber-go/zap/blob/master/FAQ.md#does-zap-support-log-rotation).

However, it seems zap must log the message string as a value with `MessageKey` as per its [EncoderConfig](https://pkg.go.dev/go.uber.org/zap/zapcore#EncoderConfig). Then, I tried to unmarshal the nested JSON to a nested struct (ex. outer struct containing [this struct](https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/webservice/state_dump.go#L51)). Problem with this is that the logged message is still surrounded by `"`, so it doesn't unmarshal in to the appropriate type (only string).

Happy to take another stab at it if you have any pointers, or want me to send a commit with screenshots of the zap log output to get a better idea of the issue. Functionally, things are working fine as per [YUNIKORN-949](https://issues.apache.org/jira/browse/YUNIKORN-949).